terraso_backend/apps/project_management/models/projects.py

```diff
@@ -0,0 +1,59 @@
+# Copyright © 2023 Technology Matters
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see https://www.gnu.org/licenses/.
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+from apps.core import permission_rules
+from apps.core.models import Group, User
+from apps.core.models.commons import BaseModel
+
+
+class Project(BaseModel):
+    class Meta(BaseModel.Meta):
+        abstract = False
+
+        rules_permissions = {"change": permission_rules.allowed_to_change_project}
+
+    PRIVATE = "private"
+    PUBLIC = "public"
+    DEFAULT_PRIVACY_STATUS = PRIVATE
+
+    PRIVACY_STATUS = ((PRIVATE, _("Private")), (PUBLIC, _("Public")))
+
+    name = models.CharField(max_length=200)
+    group = models.OneToOneField(Group, on_delete=models.CASCADE)
+    privacy = models.CharField(
```

Comment:
Can you elaborate a bit Jose? I don't think I understand.

Reply:
@josebui i think the distinction is that `open`, `closed`, and `restricted` are essentially edit permissions of the group membership (who can add a new member), whereas `private`/`public` is a view permission for non-members of the group. so if we keep `open`, `closed`, etc at the membership group level we can avoid duplicating the associated auth logic

Reply:
@shrouxm The requiement for `closed` and `open` groups orginated from the need to make data within a group not accessible by non-members, so this was not related to how a user can join, both right now can be joined directly or joined direclty pending for an approval.

Also, if we keep `closed` and `restricted` as group type that distinction from the name is not clear, so those names by themselve are not telling us a difference, because IMO they are being used to group different sets of requirements, if we split them (see my reply in the previous comment) we can make this more flexible

Reply:
@david-code What I mean (sorry I am repeating a little what was said in the previous comment) is that this concepts of private, public, closed and open are overlapping through common code, and that is confusing since we are planning to split the membership logic we have to identify what is relevant to 1000L group, landspace and project, and what is relevant to the membership entity. How the data within a landscape, 1000L group or project is accesed seems like logic not related to membership, from this example even the naming from the bussiness point of view changes, 1000L groups can be open or close and projects and be public or private, so this concept IMO should not be part of the membership logic

Reply:
Got it Jose, thanks for clarifying!